config/device-classes/generic/procurve.yaml

```
@@ -0,0 +1,47 @@
+name: procurve
+
+config:
+  components:
+    cpu: true
+    memory: true
+
+match:
+  logical_operator: "OR"
+  conditions:
+    - type: SysObjectID
+      match_mode: startsWith
+      values:
+        - ".1.3.6.1.4.1.11.2.3.7.11"
+
+identify:
+  properties:
+    vendor:
+      - detection: constant
+        value: "HP ProCurve"
+    model:
+      - detection: SysObjectID
+        operators:
+          - type: modify
+            modify_method: map
+            mappings: sysObjectID.yaml
+    serial_number:
+      - detection: snmpget
+        oid: .1.3.6.1.2.1.47.1.1.1.1.11.1001
+    os_version:
+      - detection: SysDescription
+        operators:
+          - type: modify
+            modify_method: regexSubmatch
+            regex: 'revision ([^,]+),'
+            format: "$1"
+
+components:
+  cpu:
+    load:
+      - detection: snmpget
+        oid: .1.3.6.1.4.1.11.2.14.11.5.1.9.6.1.0
+  memory:
+    usage:
+      - detection: snmpget
+        oid: .1.3.6.1.4.1.11.2.14.11.5.1.1.2.1.1.1.6.1
```

This OID returns the available memory in Bytes, but we need the percentage.
```suggestion
        oid: .1.3.6.1.4.1.11.2.14.11.5.1.1.2.1.1.1.7.1
        operators:
          - type: modify
            modify_method: divide
            value:
              detection: snmpget
              oid: 1.3.6.1.4.1.11.2.14.11.5.1.1.2.1.1.1.5.1
              operators:
                - type: modify
                  modify_method: divide
                  value:
                    detection: constant
                    value: 100
```